src/chips/hotel/src/chip.rs

```diff
@@ -27,11 +28,19 @@ impl Chip for Hotel {
 
     fn service_pending_interrupts(&mut self) {
         unsafe {
-            cortexm3::nvic::next_pending().map(|nvic_num| {
-                match nvic_num {
-                    _   => {}
+            loop {
+                match cortexm3::nvic::next_pending() {
+                    Some(nvic_num) => {
```

Comment:
It would need to return an iterator, wouldn't it? Or does it implicitly work for something that returns an `Option`?

Either way, not a bad idea